### PR TITLE
Change format of NonEmptyVector.show

### PR DESCRIPTION
**Pull request by ceedubs**
This follows up on several small suggestions from comments in #1137. The
only meaningful modification is changing the format of `show` from
something like `NonEmptyVector(Vector(1, 2, 3))` to something like
`NonEmptyVector(1, 2, 3)`.

I've also added a `length` method to `NonEmptyVector`.

### CODE REVIEW

**Comment by johnynek** [comment]
👍

File: core/src/main/scala/cats/data/NonEmptyVector.scala [hunk 1]
```diff
@@ -138,7 +144,7 @@ private[data] sealed trait NonEmptyVectorInstances {
 
       override def split[A](fa: NonEmptyVector[A]): (A, Vector[A]) = (fa.head, fa.tail)
 
-      override def size[A](fa: NonEmptyVector[A]): Long = 1 + fa.tail.size.toLong
+      override def size[A](fa: NonEmptyVector[A]): Long = fa.length.toLong
```
Comment (kailuowang) [inline core/src/main/scala/cats/data/NonEmptyVector.scala]: good catch there.

**Comment by kailuowang** [comment]
👍  LGTM pending the `tut` fix.

**Comment by non** [comment]
👍

**Comment by peterneyens** [comment]
:+1:

**Pull request merged**